frontend/rust-lib/flowy-document2/tests/document/document_test.rs

```diff
@@ -0,0 +1,210 @@
+use std::{collections::HashMap, sync::Arc, vec};
+
+use collab_document::blocks::{Block, BlockAction, BlockActionPayload, BlockActionType};
+use flowy_document2::{document::DocumentDataWrapper, manager::DocumentManager};
+use nanoid::nanoid;
+use serde_json::{json, to_value, Value};
+
+use super::util::FakeUser;
+
+#[test]
+fn restore_document() {
+  let user = FakeUser::new();
+  let manager = DocumentManager::new(Arc::new(user));
+
+  // create a document
+  let doc_id: String = nanoid!(10);
+  let data = DocumentDataWrapper::default();
+  let document_a = manager
+    .create_document(doc_id.clone(), data.clone())
+    .unwrap();
+  let data_a = document_a.lock().get_document().unwrap();
+  assert_eq!(data_a, data.0);
+
+  // open a document
+  let data_b = manager
+    .open_document(doc_id.clone())
+    .unwrap()
+    .lock()
+    .get_document()
+    .unwrap();
+  // close a document
+  _ = manager.close_document(doc_id.clone());
+  assert_eq!(data_b, data.0);
+
+  // restore
+  _ = manager.create_document(doc_id.clone(), data.clone());
+  // open a document
+  let data_b = manager
+    .open_document(doc_id.clone())
+    .unwrap()
+    .lock()
+    .get_document()
+    .unwrap();
+  // close a document
+  _ = manager.close_document(doc_id.clone());
+
+  assert_eq!(data_b, data.0);
+}
+
+#[test]
+fn document_apply_insert_action() {
+  let user = FakeUser::new();
+  let manager = DocumentManager::new(Arc::new(user));
+
+  let doc_id: String = nanoid!(10);
+  let data = DocumentDataWrapper::default();
+
+  // create a document
+  _ = manager.create_document(doc_id.clone(), data.clone());
+
+  // open a document
+  let document = manager.open_document(doc_id.clone()).unwrap();
+  let page_block = document.lock().get_block(&data.0.page_id).unwrap();
+
+  // insert a text block
+  let text_block = Block {
+    id: nanoid!(10),
+    ty: "text".to_string(),
```

Comment:
I'm a bit shocked that someone got away with saving 2 characters to finish this and make it `type`... 😂

Reply:
type is the keyword. :-D
https://doc.rust-lang.org/std/keyword.type.html

Reply:
Haha alright, I was so confused when I saw it in flutter objects 🤣